// weaviate/src/api/ModelManifestsApi.js
/**
 * Weaviate API
 * Lets you register, view and manage cloud ready devices.
 *
 * OpenAPI spec version: v1-alpha
 * 
 *
 * NOTE: This class is auto generated by the swagger code generator program.
 * https://github.com/swagger-api/swagger-codegen.git
 * Do not edit the class manually.
 *
 */

(function(root, factory) {
  if (typeof define === 'function' && define.amd) {
    // AMD. Register as an anonymous module.
    define(['ApiClient', 'model/ModelManifest', 'model/ModelManifestsListResponse', 'model/ModelManifestsValidateCommandDefsRequest', 'model/ModelManifestsValidateCommandDefsResponse', 'model/ModelManifestsValidateComponentsRequest', 'model/ModelManifestsValidateComponentsResponse', 'model/ModelManifestsValidateDeviceStateRequest', 'model/ModelManifestsValidateDeviceStateResponse'], factory);
  } else if (typeof module === 'object' && module.exports) {
    // CommonJS-like environments that support module.exports, like Node.
    module.exports = factory(require('../ApiClient'), require('../model/ModelManifest'), require('../model/ModelManifestsListResponse'), require('../model/ModelManifestsValidateCommandDefsRequest'), require('../model/ModelManifestsValidateCommandDefsResponse'), require('../model/ModelManifestsValidateComponentsRequest'), require('../model/ModelManifestsValidateComponentsResponse'), require('../model/ModelManifestsValidateDeviceStateRequest'), require('../model/ModelManifestsValidateDeviceStateResponse'));
  } else {
    // Browser globals (root is window)
    if (!root.WeaviateApi) {
      root.WeaviateApi = {};
    }
    root.WeaviateApi.ModelManifestsApi = factory(root.WeaviateApi.ApiClient, root.WeaviateApi.ModelManifest, root.WeaviateApi.ModelManifestsListResponse, root.WeaviateApi.ModelManifestsValidateCommandDefsRequest, root.WeaviateApi.ModelManifestsValidateCommandDefsResponse, root.WeaviateApi.ModelManifestsValidateComponentsRequest, root.WeaviateApi.ModelManifestsValidateComponentsResponse, root.WeaviateApi.ModelManifestsValidateDeviceStateRequest, root.WeaviateApi.ModelManifestsValidateDeviceStateResponse);
  }
}(this, function(ApiClient, ModelManifest, ModelManifestsListResponse, ModelManifestsValidateCommandDefsRequest, ModelManifestsValidateCommandDefsResponse, ModelManifestsValidateComponentsRequest, ModelManifestsValidateComponentsResponse, ModelManifestsValidateDeviceStateRequest, ModelManifestsValidateDeviceStateResponse) {
  'use strict';

  /**
   * ModelManifests service.
   * @module api/ModelManifestsApi
   * @version v1-alpha
   */

  /**
   * Constructs a new ModelManifestsApi. 
   * @alias module:api/ModelManifestsApi
   * @class
   * @param {module:ApiClient} apiClient Optional API client implementation to use,
   * default to {@link module:ApiClient#instance} if unspecified.
   */
  var exports = function(apiClient) {
    this.apiClient = apiClient || ApiClient.instance;


    /**
     * Callback function to receive the result of the weaveModelManifestsGet operation.
     * @callback module:api/ModelManifestsApi~weaveModelManifestsGetCallback
     * @param {String} error Error message, if any.
     * @param {module:model/ModelManifest} data The data returned by the service call.
     * @param {String} response The complete HTTP response.
     */

    /**
     * Returns a particular model manifest.
     * @param {String} modelManifestId Unique ID of the model manifest.
     * @param {Object} opts Optional parameters
     * @param {module:model/String} opts.alt Data format for the response. (default to json)
     * @param {String} opts.fields Selector specifying which fields to include in a partial response.
     * @param {String} opts.key API key. Your API key identifies your project and provides you with API access, quota, and reports. Required unless you provide an OAuth 2.0 token.
     * @param {String} opts.oauthToken OAuth 2.0 token for the current user.
     * @param {Boolean} opts.prettyPrint Returns response with indentations and line breaks. (default to true)
     * @param {String} opts.quotaUser Available to use for quota purposes for server-side applications. Can be any arbitrary string assigned to a user, but should not exceed 40 characters. Overrides userIp if both are provided.
     * @param {String} opts.userIp IP address of the site where the request originates. Use this if you want to enforce per-user limits.
     * @param {String} opts.hl Specifies the language code that should be used for text values in the API response.
     * @param {module:api/ModelManifestsApi~weaveModelManifestsGetCallback} callback The callback function, accepting three arguments: error, data, response
     * data is of type: {@link module:model/ModelManifest}
     */
    this.weaveModelManifestsGet = function(modelManifestId, opts, callback) {
      opts = opts || {};
      var postBody = null;

      // verify the required parameter 'modelManifestId' is set
      if (modelManifestId == undefined || modelManifestId == null) {
        throw new Error("Missing the required parameter 'modelManifestId' when calling weaveModelManifestsGet");
      }


      var pathParams = {
        'modelManifestId': modelManifestId
      };
      var queryParams = {
        'alt': opts['alt'],
        'fields': opts['fields'],
        'key': opts['key'],
        'oauth_token': opts['oauthToken'],
        'prettyPrint': opts['prettyPrint'],
        'quotaUser': opts['quotaUser'],
        'userIp': opts['userIp'],
        'hl': opts['hl']
      };
      var headerParams = {
      };
      var formParams = {
      };

      var authNames = ['apiKey'];
      var contentTypes = ['application/json', 'application/protobuf', 'application/xml'];
      var accepts = ['application/json', 'application/protobuf', 'application/xml'];
      var returnType = ModelManifest;

      return this.apiClient.callApi(
        '/modelManifests/{modelManifestId}', 'GET',
        pathParams, queryParams, headerParams, formParams, postBody,
        authNames, contentTypes, accepts, returnType, callback
      );
    }

    /**
     * Callback function to receive the result of the weaveModelManifestsList operation.
     * @callback module:api/ModelManifestsApi~weaveModelManifestsListCallback
     * @param {String} error Error message, if any.
     * @param {module:model/ModelManifestsListResponse} data The data returned by the service call.
     * @param {String} response The complete HTTP response.
     */

    /**
     * Lists all model manifests.
     * @param {Array.<String>} ids Model manifest IDs to include in the result
     * @param {Object} opts Optional parameters
     * @param {module:model/String} opts.alt Data format for the response. (default to json)
     * @param {String} opts.fields Selector specifying which fields to include in a partial response.
     * @param {String} opts.key API key. Your API key identifies your project and provides you with API access, quota, and reports. Required unless you provide an OAuth 2.0 token.
     * @param {String} opts.oauthToken OAuth 2.0 token for the current user.
     * @param {Boolean} opts.prettyPrint Returns response with indentations and line breaks. (default to true)
     * @param {String} opts.quotaUser Available to use for quota purposes for server-side applications. Can be any arbitrary string assigned to a user, but should not exceed 40 characters. Overrides userIp if both are provided.
     * @param {String} opts.userIp IP address of the site where the request originates. Use this if you want to enforce per-user limits.
     * @param {String} opts.hl Specifies the language code that should be used for text values in the API response.
     * @param {Number} opts.maxResults 
     * @param {Number} opts.startIndex 
     * @param {String} opts.token 
     * @param {module:api/ModelManifestsApi~weaveModelManifestsListCallback} callback The callback function, accepting three arguments: error, data, response
     * data is of type: {@link module:model/ModelManifestsListResponse}
     */
    this.weaveModelManifestsList = function(ids, opts, callback) {
      opts = opts || {};
      var postBody = null;

      // verify the required parameter 'ids' is set
      if (ids == undefined || ids == null) {
        throw new Error("Missing the required parameter 'ids' when calling weaveModelManifestsList");
      }


      var pathParams = {
      };
      var queryParams = {
        'alt': opts['alt'],
        'fields': opts['fields'],
        'key': opts['key'],
        'oauth_token': opts['oauthToken'],
        'prettyPrint': opts['prettyPrint'],
        'quotaUser': opts['quotaUser'],
        'userIp': opts['userIp'],
        'ids': this.apiClient.buildCollectionParam(ids, 'multi'),
        'hl': opts['hl'],
        'maxResults': opts['maxResults'],
        'startIndex': opts['startIndex'],
        'token': opts['token']
      };
      var headerParams = {
      };
      var formParams = {
      };

      var authNames = ['apiKey'];
      var contentTypes = ['application/json', 'application/protobuf', 'application/xml'];
      var accepts = ['application/json', 'application/protobuf', 'application/xml'];
      var returnType = ModelManifestsListResponse;

      return this.apiClient.callApi(
        '/modelManifests', 'GET',
        pathParams, queryParams, headerParams, formParams, postBody,
        authNames, contentTypes, accepts, returnType, callback
      );
    }

    /**
     * Callback function to receive the result of the weaveModelManifestsValidateCommandDefs operation.
     * @callback module:api/ModelManifestsApi~weaveModelManifestsValidateCommandDefsCallback
     * @param {String} error Error message, if any.
     * @param {module:model/ModelManifestsValidateCommandDefsResponse} data The data returned by the service call.
     * @param {String} response The complete HTTP response.
     */

    /**
     * Validates given command definitions and returns errors.
     * @param {Object} opts Optional parameters
     * @param {module:model/String} opts.alt Data format for the response. (default to json)
     * @param {String} opts.fields Selector specifying which fields to include in a partial response.
     * @param {String} opts.key API key. Your API key identifies your project and provides you with API access, quota, and reports. Required unless you provide an OAuth 2.0 token.
     * @param {String} opts.oauthToken OAuth 2.0 token for the current user.
     * @param {Boolean} opts.prettyPrint Returns response with indentations and line breaks. (default to true)
     * @param {String} opts.quotaUser Available to use for quota purposes for server-side applications. Can be any arbitrary string assigned to a user, but should not exceed 40 characters. Overrides userIp if both are provided.
     * @param {String} opts.userIp IP address of the site where the request originates. Use this if you want to enforce per-user limits.
     * @param {String} opts.hl Specifies the language code that should be used for text values in the API response.
     * @param {module:model/ModelManifestsValidateCommandDefsRequest} opts.body 
     * @param {module:api/ModelManifestsApi~weaveModelManifestsValidateCommandDefsCallback} callback The callback function, accepting three arguments: error, data, response
     * data is of type: {@link module:model/ModelManifestsValidateCommandDefsResponse}
     */
    this.weaveModelManifestsValidateCommandDefs = function(opts, callback) {
      opts = opts || {};
      var postBody = opts['body'];


      var pathParams = {
      };
      var queryParams = {
        'alt': opts['alt'],
        'fields': opts['fields'],
        'key': opts['key'],
        'oauth_token': opts['oauthToken'],
        'prettyPrint': opts['prettyPrint'],
        'quotaUser': opts['quotaUser'],
        'userIp': opts['userIp'],
        'hl': opts['hl']
      };
      var headerParams = {
      };
      var formParams = {
      };

      var authNames = ['apiKey'];
      var contentTypes = ['application/json', 'application/protobuf', 'application/xml'];
      var accepts = ['application/json', 'application/protobuf', 'application/xml'];
      var returnType = ModelManifestsValidateCommandDefsResponse;

      return this.apiClient.callApi(
        '/modelManifests/validateCommandDefs', 'POST',
        pathParams, queryParams, headerParams, formParams, postBody,
        authNames, contentTypes, accepts, returnType, callback
      );
    }

    /**
     * Callback function to receive the result of the weaveModelManifestsValidateComponents operation.
     * @callback module:api/ModelManifestsApi~weaveModelManifestsValidateComponentsCallback
     * @param {String} error Error message, if any.
     * @param {module:model/ModelManifestsValidateComponentsResponse} data The data returned by the service call.
     * @param {String} response The complete HTTP response.
     */

    /**
     * Validates given components definitions and returns errors.
     * @param {Object} opts Optional parameters
     * @param {module:model/String} opts.alt Data format for the response. (default to json)
     * @param {String} opts.fields Selector specifying which fields to include in a partial response.
     * @param {String} opts.key API key. Your API key identifies your project and provides you with API access, quota, and reports. Required unless you provide an OAuth 2.0 token.
     * @param {String} opts.oauthToken OAuth 2.0 token for the current user.
     * @param {Boolean} opts.prettyPrint Returns response with indentations and line breaks. (default to true)
     * @param {String} opts.quotaUser Available to use for quota purposes for server-side applications. Can be any arbitrary string assigned to a user, but should not exceed 40 characters. Overrides userIp if both are provided.
     * @param {String} opts.userIp IP address of the site where the request originates. Use this if you want to enforce per-user limits.
     * @param {String} opts.hl Specifies the language code that should be used for text values in the API response.
     * @param {module:model/ModelManifestsValidateComponentsRequest} opts.body 
     * @param {module:api/ModelManifestsApi~weaveModelManifestsValidateComponentsCallback} callback The callback function, accepting three arguments: error, data, response
     * data is of type: {@link module:model/ModelManifestsValidateComponentsResponse}
     */
    this.weaveModelManifestsValidateComponents = function(opts, callback) {
      opts = opts || {};
      var postBody = opts['body'];


      var pathParams = {
      };
      var queryParams = {
        'alt': opts['alt'],
        'fields': opts['fields'],
        'key': opts['key'],
        'oauth_token': opts['oauthToken'],
        'prettyPrint': opts['prettyPrint'],
        'quotaUser': opts['quotaUser'],
        'userIp': opts['userIp'],
        'hl': opts['hl']
      };
      var headerParams = {
      };
      var formParams = {
      };

      var authNames = ['apiKey'];
      var contentTypes = ['application/json', 'application/protobuf', 'application/xml'];
      var accepts = ['application/json', 'application/protobuf', 'application/xml'];
      var returnType = ModelManifestsValidateComponentsResponse;

      return this.apiClient.callApi(
        '/modelManifests/validateComponents', 'POST',
        pathParams, queryParams, headerParams, formParams, postBody,
        authNames, contentTypes, accepts, returnType, callback
      );
    }

    /**
     * Callback function to receive the result of the weaveModelManifestsValidateDeviceState operation.
     * @callback module:api/ModelManifestsApi~weaveModelManifestsValidateDeviceStateCallback
     * @param {String} error Error message, if any.
     * @param {module:model/ModelManifestsValidateDeviceStateResponse} data The data returned by the service call.
     * @param {String} response The complete HTTP response.
     */

    /**
     * Validates given device state object and returns errors.
     * @param {Object} opts Optional parameters
     * @param {module:model/String} opts.alt Data format for the response. (default to json)
     * @param {String} opts.fields Selector specifying which fields to include in a partial response.
     * @param {String} opts.key API key. Your API key identifies your project and provides you with API access, quota, and reports. Required unless you provide an OAuth 2.0 token.
     * @param {String} opts.oauthToken OAuth 2.0 token for the current user.
     * @param {Boolean} opts.prettyPrint Returns response with indentations and line breaks. (default to true)
     * @param {String} opts.quotaUser Available to use for quota purposes for server-side applications. Can be any arbitrary string assigned to a user, but should not exceed 40 characters. Overrides userIp if both are provided.
     * @param {String} opts.userIp IP address of the site where the request originates. Use this if you want to enforce per-user limits.
     * @param {String} opts.hl Specifies the language code that should be used for text values in the API response.
     * @param {module:model/ModelManifestsValidateDeviceStateRequest} opts.body 
     * @param {module:api/ModelManifestsApi~weaveModelManifestsValidateDeviceStateCallback} callback The callback function, accepting three arguments: error, data, response
     * data is of type: {@link module:model/ModelManifestsValidateDeviceStateResponse}
     */
    this.weaveModelManifestsValidateDeviceState = function(opts, callback) {
      opts = opts || {};
      var postBody = opts['body'];


      var pathParams = {
      };
      var queryParams = {
        'alt': opts['alt'],
        'fields': opts['fields'],
        'key': opts['key'],
        'oauth_token': opts['oauthToken'],
        'prettyPrint': opts['prettyPrint'],
        'quotaUser': opts['quotaUser'],
        'userIp': opts['userIp'],
        'hl': opts['hl']
      };
      var headerParams = {
      };
      var formParams = {
      };

      var authNames = ['apiKey'];
      var contentTypes = ['application/json', 'application/protobuf', 'application/xml'];
      var accepts = ['application/json', 'application/protobuf', 'application/xml'];
      var returnType = ModelManifestsValidateDeviceStateResponse;

      return this.apiClient.callApi(
        '/modelManifests/validateDeviceState', 'POST',
        pathParams, queryParams, headerParams, formParams, postBody,
        authNames, contentTypes, accepts, returnType, callback
      );
    }
  };

  return exports;
}));
