// weaviate/src/model/AclEntry.js
/**
 * Weaviate API
 * Lets you register, view and manage cloud ready devices.
 *
 * OpenAPI spec version: v1-alpha
 * 
 *
 * NOTE: This class is auto generated by the swagger code generator program.
 * https://github.com/swagger-api/swagger-codegen.git
 * Do not edit the class manually.
 *
 */

(function(root, factory) {
  if (typeof define === 'function' && define.amd) {
    // AMD. Register as an anonymous module.
    define(['ApiClient', 'model/LocalAccessInfo'], factory);
  } else if (typeof module === 'object' && module.exports) {
    // CommonJS-like environments that support module.exports, like Node.
    module.exports = factory(require('../ApiClient'), require('./LocalAccessInfo'));
  } else {
    // Browser globals (root is window)
    if (!root.WeaviateApi) {
      root.WeaviateApi = {};
    }
    root.WeaviateApi.AclEntry = factory(root.WeaviateApi.ApiClient, root.WeaviateApi.LocalAccessInfo);
  }
}(this, function(ApiClient, LocalAccessInfo) {
  'use strict';




  /**
   * The AclEntry model module.
   * @module model/AclEntry
   * @version v1-alpha
   */

  /**
   * Constructs a new <code>AclEntry</code>.
   * @alias module:model/AclEntry
   * @class
   */
  var exports = function() {
    var _this = this;

















  };

  /**
   * Constructs a <code>AclEntry</code> from a plain JavaScript object, optionally creating a new instance.
   * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
   * @param {Object} data The plain JavaScript object bearing properties of interest.
   * @param {module:model/AclEntry} obj Optional instance to populate.
   * @return {module:model/AclEntry} The populated <code>AclEntry</code> instance.
   */
  exports.constructFromObject = function(data, obj) {
    if (data) {
      obj = obj || new exports();

      if (data.hasOwnProperty('cloudAccessRevoked')) {
        obj['cloudAccessRevoked'] = ApiClient.convertToType(data['cloudAccessRevoked'], 'Boolean');
      }
      if (data.hasOwnProperty('creatorEmail')) {
        obj['creatorEmail'] = ApiClient.convertToType(data['creatorEmail'], 'String');
      }
      if (data.hasOwnProperty('delegator')) {
        obj['delegator'] = ApiClient.convertToType(data['delegator'], 'String');
      }
      if (data.hasOwnProperty('id')) {
        obj['id'] = ApiClient.convertToType(data['id'], 'String');
      }
      if (data.hasOwnProperty('key')) {
        obj['key'] = ApiClient.convertToType(data['key'], 'String');
      }
      if (data.hasOwnProperty('kind')) {
        obj['kind'] = ApiClient.convertToType(data['kind'], 'String');
      }
      if (data.hasOwnProperty('localAccessInfo')) {
        obj['localAccessInfo'] = LocalAccessInfo.constructFromObject(data['localAccessInfo']);
      }
      if (data.hasOwnProperty('pending')) {
        obj['pending'] = ApiClient.convertToType(data['pending'], 'Boolean');
      }
      if (data.hasOwnProperty('privileges')) {
        obj['privileges'] = ApiClient.convertToType(data['privileges'], ['String']);
      }
      if (data.hasOwnProperty('revocationTimeMs')) {
        obj['revocationTimeMs'] = ApiClient.convertToType(data['revocationTimeMs'], 'String');
      }
      if (data.hasOwnProperty('role')) {
        obj['role'] = ApiClient.convertToType(data['role'], 'String');
      }
      if (data.hasOwnProperty('scopeId')) {
        obj['scopeId'] = ApiClient.convertToType(data['scopeId'], 'String');
      }
      if (data.hasOwnProperty('scopeMembership')) {
        obj['scopeMembership'] = ApiClient.convertToType(data['scopeMembership'], 'String');
      }
      if (data.hasOwnProperty('scopeName')) {
        obj['scopeName'] = ApiClient.convertToType(data['scopeName'], 'String');
      }
      if (data.hasOwnProperty('scopePhotoUrl')) {
        obj['scopePhotoUrl'] = ApiClient.convertToType(data['scopePhotoUrl'], 'String');
      }
      if (data.hasOwnProperty('scopeType')) {
        obj['scopeType'] = ApiClient.convertToType(data['scopeType'], 'String');
      }
    }
    return obj;
  }

  /**
   * Indicates whether the AclEntry has been revoked from the cloud and the user has no cloud access, but they still might have local auth tokens that are valid and can access the device and execute commands locally. See localAccessInfo for local auth details.
   * @member {Boolean} cloudAccessRevoked
   */
  exports.prototype['cloudAccessRevoked'] = undefined;
  /**
   * User who created this entry. At the moment it is populated only when pending == true.
   * @member {String} creatorEmail
   */
  exports.prototype['creatorEmail'] = undefined;
  /**
   * User on behalf of whom the access is granted to the application.
   * @member {String} delegator
   */
  exports.prototype['delegator'] = undefined;
  /**
   * Unique ACL entry ID.
   * @member {String} id
   */
  exports.prototype['id'] = undefined;
  /**
   * Public access key value. Set only when scopeType is PUBLIC.
   * @member {String} key
   */
  exports.prototype['key'] = undefined;
  /**
   * Identifies what kind of resource this is. Value: the fixed string \"weave#aclEntry\".
   * @member {String} kind
   * @default 'weave#aclEntry'
   */
  exports.prototype['kind'] = 'weave#aclEntry';
  /**
   * @member {module:model/LocalAccessInfo} localAccessInfo
   */
  exports.prototype['localAccessInfo'] = undefined;
  /**
   * Whether this ACL entry is pending for user reply to accept/reject it.
   * @member {Boolean} pending
   */
  exports.prototype['pending'] = undefined;
  /**
   * Set of access privileges granted for this scope.  Valid values are:   - \"modifyAcl\"  - \"viewAllEvents\"
   * @member {Array.<module:model/AclEntry.PrivilegesEnum>} privileges
   */
  exports.prototype['privileges'] = undefined;
  /**
   * Time in milliseconds since Unix Epoch indicating when the AclEntry was revoked.
   * @member {String} revocationTimeMs
   */
  exports.prototype['revocationTimeMs'] = undefined;
  /**
   * Access role granted to this scope.
   * @member {module:model/AclEntry.RoleEnum} role
   */
  exports.prototype['role'] = undefined;
  /**
   * Email address if scope type is user or group, domain name if scope type is a domain.
   * @member {String} scopeId
   */
  exports.prototype['scopeId'] = undefined;
  /**
   * Type of membership the user has in the scope.
   * @member {module:model/AclEntry.ScopeMembershipEnum} scopeMembership
   */
  exports.prototype['scopeMembership'] = undefined;
  /**
   * Displayable scope name.
   * @member {String} scopeName
   */
  exports.prototype['scopeName'] = undefined;
  /**
   * URL of this scope displayable photo.
   * @member {String} scopePhotoUrl
   */
  exports.prototype['scopePhotoUrl'] = undefined;
  /**
   * Type of the access scope.
   * @member {module:model/AclEntry.ScopeTypeEnum} scopeType
   */
  exports.prototype['scopeType'] = undefined;


  /**
   * Allowed values for the <code>privileges</code> property.
   * @enum {String}
   * @readonly
   */
  exports.PrivilegesEnum = {
    /**
     * value: "modifyAcl"
     * @const
     */
    "modifyAcl": "modifyAcl",
    /**
     * value: "viewAllEvents"
     * @const
     */
    "viewAllEvents": "viewAllEvents"  };

  /**
   * Allowed values for the <code>role</code> property.
   * @enum {String}
   * @readonly
   */
  exports.RoleEnum = {
    /**
     * value: "manager"
     * @const
     */
    "manager": "manager",
    /**
     * value: "owner"
     * @const
     */
    "owner": "owner",
    /**
     * value: "robot"
     * @const
     */
    "robot": "robot",
    /**
     * value: "user"
     * @const
     */
    "user": "user",
    /**
     * value: "viewer"
     * @const
     */
    "viewer": "viewer"  };

  /**
   * Allowed values for the <code>scopeMembership</code> property.
   * @enum {String}
   * @readonly
   */
  exports.ScopeMembershipEnum = {
    /**
     * value: "delegator"
     * @const
     */
    "delegator": "delegator",
    /**
     * value: "manager"
     * @const
     */
    "manager": "manager",
    /**
     * value: "member"
     * @const
     */
    "member": "member",
    /**
     * value: "none"
     * @const
     */
    "none": "none"  };

  /**
   * Allowed values for the <code>scopeType</code> property.
   * @enum {String}
   * @readonly
   */
  exports.ScopeTypeEnum = {
    /**
     * value: "application"
     * @const
     */
    "application": "application",
    /**
     * value: "domain"
     * @const
     */
    "domain": "domain",
    /**
     * value: "group"
     * @const
     */
    "group": "group",
    /**
     * value: "public"
     * @const
     */
    "public": "public",
    /**
     * value: "user"
     * @const
     */
    "user": "user"  };


  return exports;
}));
