// weaviate/src/model/AdaptersActivateResponse.js
/**
 * Weaviate API
 * Lets you register, view and manage cloud ready devices.
 *
 * OpenAPI spec version: v1-alpha
 * 
 *
 * NOTE: This class is auto generated by the swagger code generator program.
 * https://github.com/swagger-api/swagger-codegen.git
 * Do not edit the class manually.
 *
 */

(function(root, factory) {
  if (typeof define === 'function' && define.amd) {
    // AMD. Register as an anonymous module.
    define(['ApiClient'], factory);
  } else if (typeof module === 'object' && module.exports) {
    // CommonJS-like environments that support module.exports, like Node.
    module.exports = factory(require('../ApiClient'));
  } else {
    // Browser globals (root is window)
    if (!root.WeaviateApi) {
      root.WeaviateApi = {};
    }
    root.WeaviateApi.AdaptersActivateResponse = factory(root.WeaviateApi.ApiClient);
  }
}(this, function(ApiClient) {
  'use strict';




  /**
   * The AdaptersActivateResponse model module.
   * @module model/AdaptersActivateResponse
   * @version v1-alpha
   */

  /**
   * Constructs a new <code>AdaptersActivateResponse</code>.
   * @alias module:model/AdaptersActivateResponse
   * @class
   */
  var exports = function() {
    var _this = this;




  };

  /**
   * Constructs a <code>AdaptersActivateResponse</code> from a plain JavaScript object, optionally creating a new instance.
   * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
   * @param {Object} data The plain JavaScript object bearing properties of interest.
   * @param {module:model/AdaptersActivateResponse} obj Optional instance to populate.
   * @return {module:model/AdaptersActivateResponse} The populated <code>AdaptersActivateResponse</code> instance.
   */
  exports.constructFromObject = function(data, obj) {
    if (data) {
      obj = obj || new exports();

      if (data.hasOwnProperty('activationId')) {
        obj['activationId'] = ApiClient.convertToType(data['activationId'], 'String');
      }
      if (data.hasOwnProperty('activationUrl')) {
        obj['activationUrl'] = ApiClient.convertToType(data['activationUrl'], 'String');
      }
      if (data.hasOwnProperty('kind')) {
        obj['kind'] = ApiClient.convertToType(data['kind'], 'String');
      }
    }
    return obj;
  }

  /**
   * An ID that represents the link between the adapter and the user. The activationUrl will give this ID to the adapter for use in the accept API.
   * @member {String} activationId
   */
  exports.prototype['activationId'] = undefined;
  /**
   * A URL to the adapter where the user should go to complete the activation. The URL contains the activationId and the user's email address.
   * @member {String} activationUrl
   */
  exports.prototype['activationUrl'] = undefined;
  /**
   * Identifies what kind of resource this is. Value: the fixed string \"weave#adaptersActivateResponse\".
   * @member {String} kind
   * @default 'weave#adaptersActivateResponse'
   */
  exports.prototype['kind'] = 'weave#adaptersActivateResponse';



  return exports;
}));
