// weaviate/src/model/Application.js
/**
 * Weaviate API
 * Lets you register, view and manage cloud ready devices.
 *
 * OpenAPI spec version: v1-alpha
 * 
 *
 * NOTE: This class is auto generated by the swagger code generator program.
 * https://github.com/swagger-api/swagger-codegen.git
 * Do not edit the class manually.
 *
 */

(function(root, factory) {
  if (typeof define === 'function' && define.amd) {
    // AMD. Register as an anonymous module.
    define(['ApiClient'], factory);
  } else if (typeof module === 'object' && module.exports) {
    // CommonJS-like environments that support module.exports, like Node.
    module.exports = factory(require('../ApiClient'));
  } else {
    // Browser globals (root is window)
    if (!root.WeaviateApi) {
      root.WeaviateApi = {};
    }
    root.WeaviateApi.Application = factory(root.WeaviateApi.ApiClient);
  }
}(this, function(ApiClient) {
  'use strict';




  /**
   * The Application model module.
   * @module model/Application
   * @version v1-alpha
   */

  /**
   * Constructs a new <code>Application</code>.
   * Contains information about a recommended application for a device model.
   * @alias module:model/Application
   * @class
   */
  var exports = function() {
    var _this = this;









  };

  /**
   * Constructs a <code>Application</code> from a plain JavaScript object, optionally creating a new instance.
   * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
   * @param {Object} data The plain JavaScript object bearing properties of interest.
   * @param {module:model/Application} obj Optional instance to populate.
   * @return {module:model/Application} The populated <code>Application</code> instance.
   */
  exports.constructFromObject = function(data, obj) {
    if (data) {
      obj = obj || new exports();

      if (data.hasOwnProperty('description')) {
        obj['description'] = ApiClient.convertToType(data['description'], 'String');
      }
      if (data.hasOwnProperty('iconUrl')) {
        obj['iconUrl'] = ApiClient.convertToType(data['iconUrl'], 'String');
      }
      if (data.hasOwnProperty('id')) {
        obj['id'] = ApiClient.convertToType(data['id'], 'String');
      }
      if (data.hasOwnProperty('name')) {
        obj['name'] = ApiClient.convertToType(data['name'], 'String');
      }
      if (data.hasOwnProperty('price')) {
        obj['price'] = ApiClient.convertToType(data['price'], 'Number');
      }
      if (data.hasOwnProperty('publisherName')) {
        obj['publisherName'] = ApiClient.convertToType(data['publisherName'], 'String');
      }
      if (data.hasOwnProperty('type')) {
        obj['type'] = ApiClient.convertToType(data['type'], 'String');
      }
      if (data.hasOwnProperty('url')) {
        obj['url'] = ApiClient.convertToType(data['url'], 'String');
      }
    }
    return obj;
  }

  /**
   * User readable application description.
   * @member {String} description
   */
  exports.prototype['description'] = undefined;
  /**
   * Application icon URL.
   * @member {String} iconUrl
   */
  exports.prototype['iconUrl'] = undefined;
  /**
   * Unique application ID.
   * @member {String} id
   */
  exports.prototype['id'] = undefined;
  /**
   * User readable application name.
   * @member {String} name
   */
  exports.prototype['name'] = undefined;
  /**
   * Price of the application.
   * @member {Number} price
   */
  exports.prototype['price'] = undefined;
  /**
   * User readable publisher name.
   * @member {String} publisherName
   */
  exports.prototype['publisherName'] = undefined;
  /**
   * Application type.
   * @member {module:model/Application.TypeEnum} type
   */
  exports.prototype['type'] = undefined;
  /**
   * Application install URL.
   * @member {String} url
   */
  exports.prototype['url'] = undefined;


  /**
   * Allowed values for the <code>type</code> property.
   * @enum {String}
   * @readonly
   */
  exports.TypeEnum = {
    /**
     * value: "android"
     * @const
     */
    "android": "android",
    /**
     * value: "chrome"
     * @const
     */
    "chrome": "chrome",
    /**
     * value: "ios"
     * @const
     */
    "ios": "ios",
    /**
     * value: "web"
     * @const
     */
    "web": "web"  };


  return exports;
}));
