// weaviate/src/model/Command.js
/**
 * Weaviate API
 * Lets you register, view and manage cloud ready devices.
 *
 * OpenAPI spec version: v1-alpha
 * 
 *
 * NOTE: This class is auto generated by the swagger code generator program.
 * https://github.com/swagger-api/swagger-codegen.git
 * Do not edit the class manually.
 *
 */

(function(root, factory) {
  if (typeof define === 'function' && define.amd) {
    // AMD. Register as an anonymous module.
    define(['ApiClient', 'model/CommandError', 'model/JsonObject'], factory);
  } else if (typeof module === 'object' && module.exports) {
    // CommonJS-like environments that support module.exports, like Node.
    module.exports = factory(require('../ApiClient'), require('./CommandError'), require('./JsonObject'));
  } else {
    // Browser globals (root is window)
    if (!root.WeaviateApi) {
      root.WeaviateApi = {};
    }
    root.WeaviateApi.Command = factory(root.WeaviateApi.ApiClient, root.WeaviateApi.CommandError, root.WeaviateApi.JsonObject);
  }
}(this, function(ApiClient, CommandError, JsonObject) {
  'use strict';




  /**
   * The Command model module.
   * @module model/Command
   * @version v1-alpha
   */

  /**
   * Constructs a new <code>Command</code>.
   * @alias module:model/Command
   * @class
   */
  var exports = function() {
    var _this = this;



















  };

  /**
   * Constructs a <code>Command</code> from a plain JavaScript object, optionally creating a new instance.
   * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
   * @param {Object} data The plain JavaScript object bearing properties of interest.
   * @param {module:model/Command} obj Optional instance to populate.
   * @return {module:model/Command} The populated <code>Command</code> instance.
   */
  exports.constructFromObject = function(data, obj) {
    if (data) {
      obj = obj || new exports();

      if (data.hasOwnProperty('blobParameters')) {
        obj['blobParameters'] = JsonObject.constructFromObject(data['blobParameters']);
      }
      if (data.hasOwnProperty('blobResults')) {
        obj['blobResults'] = JsonObject.constructFromObject(data['blobResults']);
      }
      if (data.hasOwnProperty('component')) {
        obj['component'] = ApiClient.convertToType(data['component'], 'String');
      }
      if (data.hasOwnProperty('creationTimeMs')) {
        obj['creationTimeMs'] = ApiClient.convertToType(data['creationTimeMs'], 'String');
      }
      if (data.hasOwnProperty('creatorEmail')) {
        obj['creatorEmail'] = ApiClient.convertToType(data['creatorEmail'], 'String');
      }
      if (data.hasOwnProperty('deviceId')) {
        obj['deviceId'] = ApiClient.convertToType(data['deviceId'], 'String');
      }
      if (data.hasOwnProperty('error')) {
        obj['error'] = CommandError.constructFromObject(data['error']);
      }
      if (data.hasOwnProperty('expirationTimeMs')) {
        obj['expirationTimeMs'] = ApiClient.convertToType(data['expirationTimeMs'], 'String');
      }
      if (data.hasOwnProperty('expirationTimeoutMs')) {
        obj['expirationTimeoutMs'] = ApiClient.convertToType(data['expirationTimeoutMs'], 'String');
      }
      if (data.hasOwnProperty('id')) {
        obj['id'] = ApiClient.convertToType(data['id'], 'String');
      }
      if (data.hasOwnProperty('kind')) {
        obj['kind'] = ApiClient.convertToType(data['kind'], 'String');
      }
      if (data.hasOwnProperty('lastUpdateTimeMs')) {
        obj['lastUpdateTimeMs'] = ApiClient.convertToType(data['lastUpdateTimeMs'], 'String');
      }
      if (data.hasOwnProperty('name')) {
        obj['name'] = ApiClient.convertToType(data['name'], 'String');
      }
      if (data.hasOwnProperty('parameters')) {
        obj['parameters'] = JsonObject.constructFromObject(data['parameters']);
      }
      if (data.hasOwnProperty('progress')) {
        obj['progress'] = JsonObject.constructFromObject(data['progress']);
      }
      if (data.hasOwnProperty('results')) {
        obj['results'] = JsonObject.constructFromObject(data['results']);
      }
      if (data.hasOwnProperty('state')) {
        obj['state'] = ApiClient.convertToType(data['state'], 'String');
      }
      if (data.hasOwnProperty('userAction')) {
        obj['userAction'] = ApiClient.convertToType(data['userAction'], 'String');
      }
    }
    return obj;
  }

  /**
   * @member {module:model/JsonObject} blobParameters
   */
  exports.prototype['blobParameters'] = undefined;
  /**
   * @member {module:model/JsonObject} blobResults
   */
  exports.prototype['blobResults'] = undefined;
  /**
   * Component name paths separated by '/'.
   * @member {String} component
   */
  exports.prototype['component'] = undefined;
  /**
   * Timestamp since epoch of a creation of a command.
   * @member {String} creationTimeMs
   */
  exports.prototype['creationTimeMs'] = undefined;
  /**
   * User that created the command (not applicable if the user is deleted).
   * @member {String} creatorEmail
   */
  exports.prototype['creatorEmail'] = undefined;
  /**
   * Device ID that this command belongs to.
   * @member {String} deviceId
   */
  exports.prototype['deviceId'] = undefined;
  /**
   * @member {module:model/CommandError} error
   */
  exports.prototype['error'] = undefined;
  /**
   * Timestamp since epoch of command expiration.
   * @member {String} expirationTimeMs
   */
  exports.prototype['expirationTimeMs'] = undefined;
  /**
   * Expiration timeout for the command since its creation, 10 seconds min, 30 days max.
   * @member {String} expirationTimeoutMs
   */
  exports.prototype['expirationTimeoutMs'] = undefined;
  /**
   * Unique command ID.
   * @member {String} id
   */
  exports.prototype['id'] = undefined;
  /**
   * Identifies what kind of resource this is. Value: the fixed string \"weave#command\".
   * @member {String} kind
   * @default 'weave#command'
   */
  exports.prototype['kind'] = 'weave#command';
  /**
   * Timestamp since epoch of last update made to the command.
   * @member {String} lastUpdateTimeMs
   */
  exports.prototype['lastUpdateTimeMs'] = undefined;
  /**
   * Full command name, including trait.
   * @member {String} name
   */
  exports.prototype['name'] = undefined;
  /**
   * @member {module:model/JsonObject} parameters
   */
  exports.prototype['parameters'] = undefined;
  /**
   * @member {module:model/JsonObject} progress
   */
  exports.prototype['progress'] = undefined;
  /**
   * @member {module:model/JsonObject} results
   */
  exports.prototype['results'] = undefined;
  /**
   * Current command state.
   * @member {module:model/Command.StateEnum} state
   */
  exports.prototype['state'] = undefined;
  /**
   * Pending command state that is not acknowledged by the device yet.
   * @member {String} userAction
   */
  exports.prototype['userAction'] = undefined;


  /**
   * Allowed values for the <code>state</code> property.
   * @enum {String}
   * @readonly
   */
  exports.StateEnum = {
    /**
     * value: "aborted"
     * @const
     */
    "aborted": "aborted",
    /**
     * value: "cancelled"
     * @const
     */
    "cancelled": "cancelled",
    /**
     * value: "done"
     * @const
     */
    "done": "done",
    /**
     * value: "error"
     * @const
     */
    "error": "error",
    /**
     * value: "expired"
     * @const
     */
    "expired": "expired",
    /**
     * value: "inProgress"
     * @const
     */
    "inProgress": "inProgress",
    /**
     * value: "queued"
     * @const
     */
    "queued": "queued"  };


  return exports;
}));
