// weaviate/src/model/CommandError.js
/**
 * Weaviate API
 * Lets you register, view and manage cloud ready devices.
 *
 * OpenAPI spec version: v1-alpha
 * 
 *
 * NOTE: This class is auto generated by the swagger code generator program.
 * https://github.com/swagger-api/swagger-codegen.git
 * Do not edit the class manually.
 *
 */

(function(root, factory) {
  if (typeof define === 'function' && define.amd) {
    // AMD. Register as an anonymous module.
    define(['ApiClient'], factory);
  } else if (typeof module === 'object' && module.exports) {
    // CommonJS-like environments that support module.exports, like Node.
    module.exports = factory(require('../ApiClient'));
  } else {
    // Browser globals (root is window)
    if (!root.WeaviateApi) {
      root.WeaviateApi = {};
    }
    root.WeaviateApi.CommandError = factory(root.WeaviateApi.ApiClient);
  }
}(this, function(ApiClient) {
  'use strict';




  /**
   * The CommandError model module.
   * @module model/CommandError
   * @version v1-alpha
   */

  /**
   * Constructs a new <code>CommandError</code>.
   * Error descriptor.
   * @alias module:model/CommandError
   * @class
   */
  var exports = function() {
    var _this = this;




  };

  /**
   * Constructs a <code>CommandError</code> from a plain JavaScript object, optionally creating a new instance.
   * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
   * @param {Object} data The plain JavaScript object bearing properties of interest.
   * @param {module:model/CommandError} obj Optional instance to populate.
   * @return {module:model/CommandError} The populated <code>CommandError</code> instance.
   */
  exports.constructFromObject = function(data, obj) {
    if (data) {
      obj = obj || new exports();

      if (data.hasOwnProperty('arguments')) {
        obj['arguments'] = ApiClient.convertToType(data['arguments'], ['String']);
      }
      if (data.hasOwnProperty('code')) {
        obj['code'] = ApiClient.convertToType(data['code'], 'String');
      }
      if (data.hasOwnProperty('message')) {
        obj['message'] = ApiClient.convertToType(data['message'], 'String');
      }
    }
    return obj;
  }

  /**
   * Positional error arguments used for error message formatting.
   * @member {Array.<String>} arguments
   */
  exports.prototype['arguments'] = undefined;
  /**
   * Error code.
   * @member {String} code
   */
  exports.prototype['code'] = undefined;
  /**
   * User-visible error message populated by the cloud based on command name and error code.
   * @member {String} message
   */
  exports.prototype['message'] = undefined;



  return exports;
}));
