// weaviate/src/model/DeviceStatePatchesStatePatch.js
/**
 * Weaviate API
 * Lets you register, view and manage cloud ready devices.
 *
 * OpenAPI spec version: v1-alpha
 * 
 *
 * NOTE: This class is auto generated by the swagger code generator program.
 * https://github.com/swagger-api/swagger-codegen.git
 * Do not edit the class manually.
 *
 */

(function(root, factory) {
  if (typeof define === 'function' && define.amd) {
    // AMD. Register as an anonymous module.
    define(['ApiClient', 'model/JsonObject'], factory);
  } else if (typeof module === 'object' && module.exports) {
    // CommonJS-like environments that support module.exports, like Node.
    module.exports = factory(require('../ApiClient'), require('./JsonObject'));
  } else {
    // Browser globals (root is window)
    if (!root.WeaviateApi) {
      root.WeaviateApi = {};
    }
    root.WeaviateApi.DeviceStatePatchesStatePatch = factory(root.WeaviateApi.ApiClient, root.WeaviateApi.JsonObject);
  }
}(this, function(ApiClient, JsonObject) {
  'use strict';




  /**
   * The DeviceStatePatchesStatePatch model module.
   * @module model/DeviceStatePatchesStatePatch
   * @version v1-alpha
   */

  /**
   * Constructs a new <code>DeviceStatePatchesStatePatch</code>.
   * Device state patch with corresponding timestamp.
   * @alias module:model/DeviceStatePatchesStatePatch
   * @class
   */
  var exports = function() {
    var _this = this;




  };

  /**
   * Constructs a <code>DeviceStatePatchesStatePatch</code> from a plain JavaScript object, optionally creating a new instance.
   * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
   * @param {Object} data The plain JavaScript object bearing properties of interest.
   * @param {module:model/DeviceStatePatchesStatePatch} obj Optional instance to populate.
   * @return {module:model/DeviceStatePatchesStatePatch} The populated <code>DeviceStatePatchesStatePatch</code> instance.
   */
  exports.constructFromObject = function(data, obj) {
    if (data) {
      obj = obj || new exports();

      if (data.hasOwnProperty('component')) {
        obj['component'] = ApiClient.convertToType(data['component'], 'String');
      }
      if (data.hasOwnProperty('patch')) {
        obj['patch'] = JsonObject.constructFromObject(data['patch']);
      }
      if (data.hasOwnProperty('timeMs')) {
        obj['timeMs'] = ApiClient.convertToType(data['timeMs'], 'String');
      }
    }
    return obj;
  }

  /**
   * Component name paths separated by '/'.
   * @member {String} component
   */
  exports.prototype['component'] = undefined;
  /**
   * @member {module:model/JsonObject} patch
   */
  exports.prototype['patch'] = undefined;
  /**
   * Timestamp of a change. Local time, UNIX timestamp or time since last boot can be used.
   * @member {String} timeMs
   */
  exports.prototype['timeMs'] = undefined;



  return exports;
}));
