// weaviate/src/model/DevicePersonalizedInfo.js
/**
 * Weaviate API
 * Lets you register, view and manage cloud ready devices.
 *
 * OpenAPI spec version: v1-alpha
 * 
 *
 * NOTE: This class is auto generated by the swagger code generator program.
 * https://github.com/swagger-api/swagger-codegen.git
 * Do not edit the class manually.
 *
 */

(function(root, factory) {
  if (typeof define === 'function' && define.amd) {
    // AMD. Register as an anonymous module.
    define(['ApiClient'], factory);
  } else if (typeof module === 'object' && module.exports) {
    // CommonJS-like environments that support module.exports, like Node.
    module.exports = factory(require('../ApiClient'));
  } else {
    // Browser globals (root is window)
    if (!root.WeaviateApi) {
      root.WeaviateApi = {};
    }
    root.WeaviateApi.DevicePersonalizedInfo = factory(root.WeaviateApi.ApiClient);
  }
}(this, function(ApiClient) {
  'use strict';




  /**
   * The DevicePersonalizedInfo model module.
   * @module model/DevicePersonalizedInfo
   * @version v1-alpha
   */

  /**
   * Constructs a new <code>DevicePersonalizedInfo</code>.
   * Personalized device information for currently logged-in user.
   * @alias module:model/DevicePersonalizedInfo
   * @class
   */
  var exports = function() {
    var _this = this;





  };

  /**
   * Constructs a <code>DevicePersonalizedInfo</code> from a plain JavaScript object, optionally creating a new instance.
   * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
   * @param {Object} data The plain JavaScript object bearing properties of interest.
   * @param {module:model/DevicePersonalizedInfo} obj Optional instance to populate.
   * @return {module:model/DevicePersonalizedInfo} The populated <code>DevicePersonalizedInfo</code> instance.
   */
  exports.constructFromObject = function(data, obj) {
    if (data) {
      obj = obj || new exports();

      if (data.hasOwnProperty('lastUseTimeMs')) {
        obj['lastUseTimeMs'] = ApiClient.convertToType(data['lastUseTimeMs'], 'String');
      }
      if (data.hasOwnProperty('location')) {
        obj['location'] = ApiClient.convertToType(data['location'], 'String');
      }
      if (data.hasOwnProperty('maxRole')) {
        obj['maxRole'] = ApiClient.convertToType(data['maxRole'], 'String');
      }
      if (data.hasOwnProperty('name')) {
        obj['name'] = ApiClient.convertToType(data['name'], 'String');
      }
    }
    return obj;
  }

  /**
   * Timestamp of the last device usage by the user in milliseconds since epoch UTC.
   * @member {String} lastUseTimeMs
   */
  exports.prototype['lastUseTimeMs'] = undefined;
  /**
   * Personalized device location.
   * @member {String} location
   */
  exports.prototype['location'] = undefined;
  /**
   * The maximum role on the device.
   * @member {String} maxRole
   */
  exports.prototype['maxRole'] = undefined;
  /**
   * Personalized device display name.
   * @member {String} name
   */
  exports.prototype['name'] = undefined;



  return exports;
}));
