// weaviate/src/model/DeviceChannelPubsub.js
/**
 * Weaviate API
 * Lets you register, view and manage cloud ready devices.
 *
 * OpenAPI spec version: v1-alpha
 * 
 *
 * NOTE: This class is auto generated by the swagger code generator program.
 * https://github.com/swagger-api/swagger-codegen.git
 * Do not edit the class manually.
 *
 */

(function(root, factory) {
  if (typeof define === 'function' && define.amd) {
    // AMD. Register as an anonymous module.
    define(['ApiClient'], factory);
  } else if (typeof module === 'object' && module.exports) {
    // CommonJS-like environments that support module.exports, like Node.
    module.exports = factory(require('../ApiClient'));
  } else {
    // Browser globals (root is window)
    if (!root.WeaviateApi) {
      root.WeaviateApi = {};
    }
    root.WeaviateApi.DeviceChannelPubsub = factory(root.WeaviateApi.ApiClient);
  }
}(this, function(ApiClient) {
  'use strict';




  /**
   * The DeviceChannelPubsub model module.
   * @module model/DeviceChannelPubsub
   * @version v1-alpha
   */

  /**
   * Constructs a new <code>DeviceChannelPubsub</code>.
   * Pubsub channel details.
   * @alias module:model/DeviceChannelPubsub
   * @class
   */
  var exports = function() {
    var _this = this;



  };

  /**
   * Constructs a <code>DeviceChannelPubsub</code> from a plain JavaScript object, optionally creating a new instance.
   * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
   * @param {Object} data The plain JavaScript object bearing properties of interest.
   * @param {module:model/DeviceChannelPubsub} obj Optional instance to populate.
   * @return {module:model/DeviceChannelPubsub} The populated <code>DeviceChannelPubsub</code> instance.
   */
  exports.constructFromObject = function(data, obj) {
    if (data) {
      obj = obj || new exports();

      if (data.hasOwnProperty('connectionStatusHint')) {
        obj['connectionStatusHint'] = ApiClient.convertToType(data['connectionStatusHint'], 'String');
      }
      if (data.hasOwnProperty('topic')) {
        obj['topic'] = ApiClient.convertToType(data['topic'], 'String');
      }
    }
    return obj;
  }

  /**
   * Device's connection status, as set by the pubsub subscriber.
   * @member {module:model/DeviceChannelPubsub.ConnectionStatusHintEnum} connectionStatusHint
   */
  exports.prototype['connectionStatusHint'] = undefined;
  /**
   * Pubsub topic to publish to with device notifications.
   * @member {String} topic
   */
  exports.prototype['topic'] = undefined;


  /**
   * Allowed values for the <code>connectionStatusHint</code> property.
   * @enum {String}
   * @readonly
   */
  exports.ConnectionStatusHintEnum = {
    /**
     * value: "offline"
     * @const
     */
    "offline": "offline",
    /**
     * value: "online"
     * @const
     */
    "online": "online",
    /**
     * value: "unknown"
     * @const
     */
    "unknown": "unknown"  };


  return exports;
}));
