// weaviate/src/model/Event.js
/**
 * Weaviate API
 * Lets you register, view and manage cloud ready devices.
 *
 * OpenAPI spec version: v1-alpha
 * 
 *
 * NOTE: This class is auto generated by the swagger code generator program.
 * https://github.com/swagger-api/swagger-codegen.git
 * Do not edit the class manually.
 *
 */

(function(root, factory) {
  if (typeof define === 'function' && define.amd) {
    // AMD. Register as an anonymous module.
    define(['ApiClient', 'model/EventCommandPatch', 'model/JsonObject'], factory);
  } else if (typeof module === 'object' && module.exports) {
    // CommonJS-like environments that support module.exports, like Node.
    module.exports = factory(require('../ApiClient'), require('./EventCommandPatch'), require('./JsonObject'));
  } else {
    // Browser globals (root is window)
    if (!root.WeaviateApi) {
      root.WeaviateApi = {};
    }
    root.WeaviateApi.Event = factory(root.WeaviateApi.ApiClient, root.WeaviateApi.EventCommandPatch, root.WeaviateApi.JsonObject);
  }
}(this, function(ApiClient, EventCommandPatch, JsonObject) {
  'use strict';




  /**
   * The Event model module.
   * @module model/Event
   * @version v1-alpha
   */

  /**
   * Constructs a new <code>Event</code>.
   * @alias module:model/Event
   * @class
   */
  var exports = function() {
    var _this = this;










  };

  /**
   * Constructs a <code>Event</code> from a plain JavaScript object, optionally creating a new instance.
   * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
   * @param {Object} data The plain JavaScript object bearing properties of interest.
   * @param {module:model/Event} obj Optional instance to populate.
   * @return {module:model/Event} The populated <code>Event</code> instance.
   */
  exports.constructFromObject = function(data, obj) {
    if (data) {
      obj = obj || new exports();

      if (data.hasOwnProperty('commandPatch')) {
        obj['commandPatch'] = EventCommandPatch.constructFromObject(data['commandPatch']);
      }
      if (data.hasOwnProperty('connectionStatus')) {
        obj['connectionStatus'] = ApiClient.convertToType(data['connectionStatus'], 'String');
      }
      if (data.hasOwnProperty('deviceId')) {
        obj['deviceId'] = ApiClient.convertToType(data['deviceId'], 'String');
      }
      if (data.hasOwnProperty('id')) {
        obj['id'] = ApiClient.convertToType(data['id'], 'String');
      }
      if (data.hasOwnProperty('kind')) {
        obj['kind'] = ApiClient.convertToType(data['kind'], 'String');
      }
      if (data.hasOwnProperty('statePatch')) {
        obj['statePatch'] = JsonObject.constructFromObject(data['statePatch']);
      }
      if (data.hasOwnProperty('timeMs')) {
        obj['timeMs'] = ApiClient.convertToType(data['timeMs'], 'String');
      }
      if (data.hasOwnProperty('type')) {
        obj['type'] = ApiClient.convertToType(data['type'], 'String');
      }
      if (data.hasOwnProperty('userEmail')) {
        obj['userEmail'] = ApiClient.convertToType(data['userEmail'], 'String');
      }
    }
    return obj;
  }

  /**
   * @member {module:model/EventCommandPatch} commandPatch
   */
  exports.prototype['commandPatch'] = undefined;
  /**
   * New device connection state (if connectivity change event).
   * @member {String} connectionStatus
   */
  exports.prototype['connectionStatus'] = undefined;
  /**
   * The device that was affected by this event.
   * @member {String} deviceId
   */
  exports.prototype['deviceId'] = undefined;
  /**
   * ID of the event.
   * @member {String} id
   */
  exports.prototype['id'] = undefined;
  /**
   * Identifies what kind of resource this is. Value: the fixed string \"weave#event\".
   * @member {String} kind
   * @default 'weave#event'
   */
  exports.prototype['kind'] = 'weave#event';
  /**
   * @member {module:model/JsonObject} statePatch
   */
  exports.prototype['statePatch'] = undefined;
  /**
   * Time the event was generated in milliseconds since epoch UTC.
   * @member {String} timeMs
   */
  exports.prototype['timeMs'] = undefined;
  /**
   * Type of the event.
   * @member {module:model/Event.TypeEnum} type
   */
  exports.prototype['type'] = undefined;
  /**
   * User that caused the event (if applicable).
   * @member {String} userEmail
   */
  exports.prototype['userEmail'] = undefined;


  /**
   * Allowed values for the <code>type</code> property.
   * @enum {String}
   * @readonly
   */
  exports.TypeEnum = {
    /**
     * value: "adapterDeactivated"
     * @const
     */
    "adapterDeactivated": "adapterDeactivated",
    /**
     * value: "commandCancelled"
     * @const
     */
    "commandCancelled": "commandCancelled",
    /**
     * value: "commandCreated"
     * @const
     */
    "commandCreated": "commandCreated",
    /**
     * value: "commandDeleted"
     * @const
     */
    "commandDeleted": "commandDeleted",
    /**
     * value: "commandExpired"
     * @const
     */
    "commandExpired": "commandExpired",
    /**
     * value: "commandUpdated"
     * @const
     */
    "commandUpdated": "commandUpdated",
    /**
     * value: "deviceAclUpdated"
     * @const
     */
    "deviceAclUpdated": "deviceAclUpdated",
    /**
     * value: "deviceConnectivityChange"
     * @const
     */
    "deviceConnectivityChange": "deviceConnectivityChange",
    /**
     * value: "deviceCreated"
     * @const
     */
    "deviceCreated": "deviceCreated",
    /**
     * value: "deviceDeleted"
     * @const
     */
    "deviceDeleted": "deviceDeleted",
    /**
     * value: "devicePlaceUpdated"
     * @const
     */
    "devicePlaceUpdated": "devicePlaceUpdated",
    /**
     * value: "deviceTransferred"
     * @const
     */
    "deviceTransferred": "deviceTransferred",
    /**
     * value: "deviceUpdated"
     * @const
     */
    "deviceUpdated": "deviceUpdated",
    /**
     * value: "deviceUseTimeUpdated"
     * @const
     */
    "deviceUseTimeUpdated": "deviceUseTimeUpdated",
    /**
     * value: "deviceUserAclCreated"
     * @const
     */
    "deviceUserAclCreated": "deviceUserAclCreated",
    /**
     * value: "deviceUserAclDeleted"
     * @const
     */
    "deviceUserAclDeleted": "deviceUserAclDeleted",
    /**
     * value: "deviceUserAclUpdated"
     * @const
     */
    "deviceUserAclUpdated": "deviceUserAclUpdated",
    /**
     * value: "eventsDeleted"
     * @const
     */
    "eventsDeleted": "eventsDeleted",
    /**
     * value: "eventsRecordingDisabled"
     * @const
     */
    "eventsRecordingDisabled": "eventsRecordingDisabled",
    /**
     * value: "eventsRecordingEnabled"
     * @const
     */
    "eventsRecordingEnabled": "eventsRecordingEnabled",
    /**
     * value: "placeCreated"
     * @const
     */
    "placeCreated": "placeCreated",
    /**
     * value: "placeDeleted"
     * @const
     */
    "placeDeleted": "placeDeleted",
    /**
     * value: "placeMemberAdded"
     * @const
     */
    "placeMemberAdded": "placeMemberAdded",
    /**
     * value: "placeMemberRemoved"
     * @const
     */
    "placeMemberRemoved": "placeMemberRemoved",
    /**
     * value: "placeUpdated"
     * @const
     */
    "placeUpdated": "placeUpdated",
    /**
     * value: "roomCreated"
     * @const
     */
    "roomCreated": "roomCreated",
    /**
     * value: "roomDeleted"
     * @const
     */
    "roomDeleted": "roomDeleted",
    /**
     * value: "roomUpdated"
     * @const
     */
    "roomUpdated": "roomUpdated"  };


  return exports;
}));
