// weaviate/src/model/EventsListResponse.js
/**
 * Weaviate API
 * Lets you register, view and manage cloud ready devices.
 *
 * OpenAPI spec version: v1-alpha
 * 
 *
 * NOTE: This class is auto generated by the swagger code generator program.
 * https://github.com/swagger-api/swagger-codegen.git
 * Do not edit the class manually.
 *
 */

(function(root, factory) {
  if (typeof define === 'function' && define.amd) {
    // AMD. Register as an anonymous module.
    define(['ApiClient', 'model/Event'], factory);
  } else if (typeof module === 'object' && module.exports) {
    // CommonJS-like environments that support module.exports, like Node.
    module.exports = factory(require('../ApiClient'), require('./Event'));
  } else {
    // Browser globals (root is window)
    if (!root.WeaviateApi) {
      root.WeaviateApi = {};
    }
    root.WeaviateApi.EventsListResponse = factory(root.WeaviateApi.ApiClient, root.WeaviateApi.Event);
  }
}(this, function(ApiClient, Event) {
  'use strict';




  /**
   * The EventsListResponse model module.
   * @module model/EventsListResponse
   * @version v1-alpha
   */

  /**
   * Constructs a new <code>EventsListResponse</code>.
   * List of events.
   * @alias module:model/EventsListResponse
   * @class
   */
  var exports = function() {
    var _this = this;





  };

  /**
   * Constructs a <code>EventsListResponse</code> from a plain JavaScript object, optionally creating a new instance.
   * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
   * @param {Object} data The plain JavaScript object bearing properties of interest.
   * @param {module:model/EventsListResponse} obj Optional instance to populate.
   * @return {module:model/EventsListResponse} The populated <code>EventsListResponse</code> instance.
   */
  exports.constructFromObject = function(data, obj) {
    if (data) {
      obj = obj || new exports();

      if (data.hasOwnProperty('events')) {
        obj['events'] = ApiClient.convertToType(data['events'], [Event]);
      }
      if (data.hasOwnProperty('kind')) {
        obj['kind'] = ApiClient.convertToType(data['kind'], 'String');
      }
      if (data.hasOwnProperty('nextPageToken')) {
        obj['nextPageToken'] = ApiClient.convertToType(data['nextPageToken'], 'String');
      }
      if (data.hasOwnProperty('totalResults')) {
        obj['totalResults'] = ApiClient.convertToType(data['totalResults'], 'Number');
      }
    }
    return obj;
  }

  /**
   * The actual list of events in reverse chronological order.
   * @member {Array.<module:model/Event>} events
   */
  exports.prototype['events'] = undefined;
  /**
   * Identifies what kind of resource this is. Value: the fixed string \"weave#eventsListResponse\".
   * @member {String} kind
   * @default 'weave#eventsListResponse'
   */
  exports.prototype['kind'] = 'weave#eventsListResponse';
  /**
   * Token for the next page of events.
   * @member {String} nextPageToken
   */
  exports.prototype['nextPageToken'] = undefined;
  /**
   * The total number of events for the query. The number of items in a response may be smaller due to paging.
   * @member {Number} totalResults
   */
  exports.prototype['totalResults'] = undefined;



  return exports;
}));
