// weaviate/src/model/LocalAccessInfo.js
/**
 * Weaviate API
 * Lets you register, view and manage cloud ready devices.
 *
 * OpenAPI spec version: v1-alpha
 * 
 *
 * NOTE: This class is auto generated by the swagger code generator program.
 * https://github.com/swagger-api/swagger-codegen.git
 * Do not edit the class manually.
 *
 */

(function(root, factory) {
  if (typeof define === 'function' && define.amd) {
    // AMD. Register as an anonymous module.
    define(['ApiClient', 'model/LocalAccessEntry'], factory);
  } else if (typeof module === 'object' && module.exports) {
    // CommonJS-like environments that support module.exports, like Node.
    module.exports = factory(require('../ApiClient'), require('./LocalAccessEntry'));
  } else {
    // Browser globals (root is window)
    if (!root.WeaviateApi) {
      root.WeaviateApi = {};
    }
    root.WeaviateApi.LocalAccessInfo = factory(root.WeaviateApi.ApiClient, root.WeaviateApi.LocalAccessEntry);
  }
}(this, function(ApiClient, LocalAccessEntry) {
  'use strict';




  /**
   * The LocalAccessInfo model module.
   * @module model/LocalAccessInfo
   * @version v1-alpha
   */

  /**
   * Constructs a new <code>LocalAccessInfo</code>.
   * @alias module:model/LocalAccessInfo
   * @class
   */
  var exports = function() {
    var _this = this;





  };

  /**
   * Constructs a <code>LocalAccessInfo</code> from a plain JavaScript object, optionally creating a new instance.
   * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
   * @param {Object} data The plain JavaScript object bearing properties of interest.
   * @param {module:model/LocalAccessInfo} obj Optional instance to populate.
   * @return {module:model/LocalAccessInfo} The populated <code>LocalAccessInfo</code> instance.
   */
  exports.constructFromObject = function(data, obj) {
    if (data) {
      obj = obj || new exports();

      if (data.hasOwnProperty('localAccessEntry')) {
        obj['localAccessEntry'] = LocalAccessEntry.constructFromObject(data['localAccessEntry']);
      }
      if (data.hasOwnProperty('localAuthTokenMintTimeMs')) {
        obj['localAuthTokenMintTimeMs'] = ApiClient.convertToType(data['localAuthTokenMintTimeMs'], 'String');
      }
      if (data.hasOwnProperty('localAuthTokenTimeLeftMs')) {
        obj['localAuthTokenTimeLeftMs'] = ApiClient.convertToType(data['localAuthTokenTimeLeftMs'], 'String');
      }
      if (data.hasOwnProperty('localAuthTokenTtlTimeMs')) {
        obj['localAuthTokenTtlTimeMs'] = ApiClient.convertToType(data['localAuthTokenTtlTimeMs'], 'String');
      }
    }
    return obj;
  }

  /**
   * @member {module:model/LocalAccessEntry} localAccessEntry
   */
  exports.prototype['localAccessEntry'] = undefined;
  /**
   * Time in milliseconds since unix epoch of when the local auth token was minted.
   * @member {String} localAuthTokenMintTimeMs
   */
  exports.prototype['localAuthTokenMintTimeMs'] = undefined;
  /**
   * Relative time left of token after API call.
   * @member {String} localAuthTokenTimeLeftMs
   */
  exports.prototype['localAuthTokenTimeLeftMs'] = undefined;
  /**
   * Time in milliseconds of hold long the token is valid after minting.
   * @member {String} localAuthTokenTtlTimeMs
   */
  exports.prototype['localAuthTokenTtlTimeMs'] = undefined;



  return exports;
}));
