// weaviate/src/model/LocalAccessEntry.js
/**
 * Weaviate API
 * Lets you register, view and manage cloud ready devices.
 *
 * OpenAPI spec version: v1-alpha
 * 
 *
 * NOTE: This class is auto generated by the swagger code generator program.
 * https://github.com/swagger-api/swagger-codegen.git
 * Do not edit the class manually.
 *
 */

(function(root, factory) {
  if (typeof define === 'function' && define.amd) {
    // AMD. Register as an anonymous module.
    define(['ApiClient'], factory);
  } else if (typeof module === 'object' && module.exports) {
    // CommonJS-like environments that support module.exports, like Node.
    module.exports = factory(require('../ApiClient'));
  } else {
    // Browser globals (root is window)
    if (!root.WeaviateApi) {
      root.WeaviateApi = {};
    }
    root.WeaviateApi.LocalAccessEntry = factory(root.WeaviateApi.ApiClient);
  }
}(this, function(ApiClient) {
  'use strict';




  /**
   * The LocalAccessEntry model module.
   * @module model/LocalAccessEntry
   * @version v1-alpha
   */

  /**
   * Constructs a new <code>LocalAccessEntry</code>.
   * @alias module:model/LocalAccessEntry
   * @class
   */
  var exports = function() {
    var _this = this;




  };

  /**
   * Constructs a <code>LocalAccessEntry</code> from a plain JavaScript object, optionally creating a new instance.
   * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
   * @param {Object} data The plain JavaScript object bearing properties of interest.
   * @param {module:model/LocalAccessEntry} obj Optional instance to populate.
   * @return {module:model/LocalAccessEntry} The populated <code>LocalAccessEntry</code> instance.
   */
  exports.constructFromObject = function(data, obj) {
    if (data) {
      obj = obj || new exports();

      if (data.hasOwnProperty('isApp')) {
        obj['isApp'] = ApiClient.convertToType(data['isApp'], 'Boolean');
      }
      if (data.hasOwnProperty('localAccessRole')) {
        obj['localAccessRole'] = ApiClient.convertToType(data['localAccessRole'], 'String');
      }
      if (data.hasOwnProperty('projectId')) {
        obj['projectId'] = ApiClient.convertToType(data['projectId'], 'String');
      }
    }
    return obj;
  }

  /**
   * Whether this belongs to a delegated app or user.
   * @member {Boolean} isApp
   */
  exports.prototype['isApp'] = undefined;
  /**
   * Access role of the user.
   * @member {module:model/LocalAccessEntry.LocalAccessRoleEnum} localAccessRole
   */
  exports.prototype['localAccessRole'] = undefined;
  /**
   * Project id of the app that this access info is associated with.
   * @member {String} projectId
   */
  exports.prototype['projectId'] = undefined;


  /**
   * Allowed values for the <code>localAccessRole</code> property.
   * @enum {String}
   * @readonly
   */
  exports.LocalAccessRoleEnum = {
    /**
     * value: "manager"
     * @const
     */
    "manager": "manager",
    /**
     * value: "owner"
     * @const
     */
    "owner": "owner",
    /**
     * value: "robot"
     * @const
     */
    "robot": "robot",
    /**
     * value: "user"
     * @const
     */
    "user": "user",
    /**
     * value: "viewer"
     * @const
     */
    "viewer": "viewer"  };


  return exports;
}));
