// weaviate/src/index.js
/**
 * Weaviate API
 * Lets you register, view and manage cloud ready devices.
 *
 * OpenAPI spec version: v1-alpha
 * 
 *
 * NOTE: This class is auto generated by the swagger code generator program.
 * https://github.com/swagger-api/swagger-codegen.git
 * Do not edit the class manually.
 *
 */

(function(factory) {
  if (typeof define === 'function' && define.amd) {
    // AMD. Register as an anonymous module.
    define(['ApiClient', 'model/AclEntriesListResponse', 'model/AclEntry', 'model/Adapter', 'model/AdaptersAcceptResponse', 'model/AdaptersActivateResponse', 'model/AdaptersDeactivateResponse', 'model/AdaptersListResponse', 'model/Application', 'model/AssociatedLabel', 'model/Command', 'model/CommandError', 'model/CommandsListResponse', 'model/CommandsQueueResponse', 'model/Device', 'model/DeviceChannel', 'model/DeviceChannelPubsub', 'model/DeviceModelManifest', 'model/DevicePersonalizedInfo', 'model/DeviceStatePatchesStatePatch', 'model/DevicesAddLabelRequest', 'model/DevicesAddNicknameRequest', 'model/DevicesAddNicknameResponse', 'model/DevicesListResponse', 'model/DevicesPatchStateRequest', 'model/DevicesPatchStateResponse', 'model/DevicesRemoveLabelRequest', 'model/DevicesRemoveNicknameRequest', 'model/DevicesRemoveNicknameResponse', 'model/Event', 'model/EventCommandPatch', 'model/EventsDeleteAllRequest', 'model/EventsListResponse', 'model/EventsRecordDeviceEventsRequest', 'model/Invitation', 'model/JsonObject', 'model/JsonValue', 'model/LocalAccessEntry', 'model/LocalAccessInfo', 'model/ModelManifest', 'model/ModelManifestsListResponse', 'model/ModelManifestsValidateCommandDefsRequest', 'model/ModelManifestsValidateCommandDefsResponse', 'model/ModelManifestsValidateComponentsRequest', 'model/ModelManifestsValidateComponentsResponse', 'model/ModelManifestsValidateDeviceStateRequest', 'model/ModelManifestsValidateDeviceStateResponse', 'model/PackageDef', 'model/PlacesHints', 'model/Room', 'model/StateDef', 'api/AclEntriesApi', 'api/AdaptersApi', 'api/CommandsApi', 'api/DevicesApi', 'api/EventsApi', 'api/ModelManifestsApi'], factory);
  } else if (typeof module === 'object' && module.exports) {
    // CommonJS-like environments that support module.exports, like Node.
    module.exports = factory(require('./ApiClient'), require('./model/AclEntriesListResponse'), require('./model/AclEntry'), require('./model/Adapter'), require('./model/AdaptersAcceptResponse'), require('./model/AdaptersActivateResponse'), require('./model/AdaptersDeactivateResponse'), require('./model/AdaptersListResponse'), require('./model/Application'), require('./model/AssociatedLabel'), require('./model/Command'), require('./model/CommandError'), require('./model/CommandsListResponse'), require('./model/CommandsQueueResponse'), require('./model/Device'), require('./model/DeviceChannel'), require('./model/DeviceChannelPubsub'), require('./model/DeviceModelManifest'), require('./model/DevicePersonalizedInfo'), require('./model/DeviceStatePatchesStatePatch'), require('./model/DevicesAddLabelRequest'), require('./model/DevicesAddNicknameRequest'), require('./model/DevicesAddNicknameResponse'), require('./model/DevicesListResponse'), require('./model/DevicesPatchStateRequest'), require('./model/DevicesPatchStateResponse'), require('./model/DevicesRemoveLabelRequest'), require('./model/DevicesRemoveNicknameRequest'), require('./model/DevicesRemoveNicknameResponse'), require('./model/Event'), require('./model/EventCommandPatch'), require('./model/EventsDeleteAllRequest'), require('./model/EventsListResponse'), require('./model/EventsRecordDeviceEventsRequest'), require('./model/Invitation'), require('./model/JsonObject'), require('./model/JsonValue'), require('./model/LocalAccessEntry'), require('./model/LocalAccessInfo'), require('./model/ModelManifest'), require('./model/ModelManifestsListResponse'), require('./model/ModelManifestsValidateCommandDefsRequest'), require('./model/ModelManifestsValidateCommandDefsResponse'), require('./model/ModelManifestsValidateComponentsRequest'), require('./model/ModelManifestsValidateComponentsResponse'), require('./model/ModelManifestsValidateDeviceStateRequest'), require('./model/ModelManifestsValidateDeviceStateResponse'), require('./model/PackageDef'), require('./model/PlacesHints'), require('./model/Room'), require('./model/StateDef'), require('./api/AclEntriesApi'), require('./api/AdaptersApi'), require('./api/CommandsApi'), require('./api/DevicesApi'), require('./api/EventsApi'), require('./api/ModelManifestsApi'));
  }
}(function(ApiClient, AclEntriesListResponse, AclEntry, Adapter, AdaptersAcceptResponse, AdaptersActivateResponse, AdaptersDeactivateResponse, AdaptersListResponse, Application, AssociatedLabel, Command, CommandError, CommandsListResponse, CommandsQueueResponse, Device, DeviceChannel, DeviceChannelPubsub, DeviceModelManifest, DevicePersonalizedInfo, DeviceStatePatchesStatePatch, DevicesAddLabelRequest, DevicesAddNicknameRequest, DevicesAddNicknameResponse, DevicesListResponse, DevicesPatchStateRequest, DevicesPatchStateResponse, DevicesRemoveLabelRequest, DevicesRemoveNicknameRequest, DevicesRemoveNicknameResponse, Event, EventCommandPatch, EventsDeleteAllRequest, EventsListResponse, EventsRecordDeviceEventsRequest, Invitation, JsonObject, JsonValue, LocalAccessEntry, LocalAccessInfo, ModelManifest, ModelManifestsListResponse, ModelManifestsValidateCommandDefsRequest, ModelManifestsValidateCommandDefsResponse, ModelManifestsValidateComponentsRequest, ModelManifestsValidateComponentsResponse, ModelManifestsValidateDeviceStateRequest, ModelManifestsValidateDeviceStateResponse, PackageDef, PlacesHints, Room, StateDef, AclEntriesApi, AdaptersApi, CommandsApi, DevicesApi, EventsApi, ModelManifestsApi) {
  'use strict';

  /**
   * Lets_you_register_view_and_manage_cloud_ready_devices_.<br>
   * The <code>index</code> module provides access to constructors for all the classes which comprise the public API.
   * <p>
   * An AMD (recommended!) or CommonJS application will generally do something equivalent to the following:
   * <pre>
   * var WeaviateApi = require('index'); // See note below*.
   * var xxxSvc = new WeaviateApi.XxxApi(); // Allocate the API class we're going to use.
   * var yyyModel = new WeaviateApi.Yyy(); // Construct a model instance.
   * yyyModel.someProperty = 'someValue';
   * ...
   * var zzz = xxxSvc.doSomething(yyyModel); // Invoke the service.
   * ...
   * </pre>
   * <em>*NOTE: For a top-level AMD script, use require(['index'], function(){...})
   * and put the application logic within the callback function.</em>
   * </p>
   * <p>
   * A non-AMD browser application (discouraged) might do something like this:
   * <pre>
   * var xxxSvc = new WeaviateApi.XxxApi(); // Allocate the API class we're going to use.
   * var yyy = new WeaviateApi.Yyy(); // Construct a model instance.
   * yyyModel.someProperty = 'someValue';
   * ...
   * var zzz = xxxSvc.doSomething(yyyModel); // Invoke the service.
   * ...
   * </pre>
   * </p>
   * @module index
   * @version v1-alpha
   */
  var exports = {
    /**
     * The ApiClient constructor.
     * @property {module:ApiClient}
     */
    ApiClient: ApiClient,
    /**
     * The AclEntriesListResponse model constructor.
     * @property {module:model/AclEntriesListResponse}
     */
    AclEntriesListResponse: AclEntriesListResponse,
    /**
     * The AclEntry model constructor.
     * @property {module:model/AclEntry}
     */
    AclEntry: AclEntry,
    /**
     * The Adapter model constructor.
     * @property {module:model/Adapter}
     */
    Adapter: Adapter,
    /**
     * The AdaptersAcceptResponse model constructor.
     * @property {module:model/AdaptersAcceptResponse}
     */
    AdaptersAcceptResponse: AdaptersAcceptResponse,
    /**
     * The AdaptersActivateResponse model constructor.
     * @property {module:model/AdaptersActivateResponse}
     */
    AdaptersActivateResponse: AdaptersActivateResponse,
    /**
     * The AdaptersDeactivateResponse model constructor.
     * @property {module:model/AdaptersDeactivateResponse}
     */
    AdaptersDeactivateResponse: AdaptersDeactivateResponse,
    /**
     * The AdaptersListResponse model constructor.
     * @property {module:model/AdaptersListResponse}
     */
    AdaptersListResponse: AdaptersListResponse,
    /**
     * The Application model constructor.
     * @property {module:model/Application}
     */
    Application: Application,
    /**
     * The AssociatedLabel model constructor.
     * @property {module:model/AssociatedLabel}
     */
    AssociatedLabel: AssociatedLabel,
    /**
     * The Command model constructor.
     * @property {module:model/Command}
     */
    Command: Command,
    /**
     * The CommandError model constructor.
     * @property {module:model/CommandError}
     */
    CommandError: CommandError,
    /**
     * The CommandsListResponse model constructor.
     * @property {module:model/CommandsListResponse}
     */
    CommandsListResponse: CommandsListResponse,
    /**
     * The CommandsQueueResponse model constructor.
     * @property {module:model/CommandsQueueResponse}
     */
    CommandsQueueResponse: CommandsQueueResponse,
    /**
     * The Device model constructor.
     * @property {module:model/Device}
     */
    Device: Device,
    /**
     * The DeviceChannel model constructor.
     * @property {module:model/DeviceChannel}
     */
    DeviceChannel: DeviceChannel,
    /**
     * The DeviceChannelPubsub model constructor.
     * @property {module:model/DeviceChannelPubsub}
     */
    DeviceChannelPubsub: DeviceChannelPubsub,
    /**
     * The DeviceModelManifest model constructor.
     * @property {module:model/DeviceModelManifest}
     */
    DeviceModelManifest: DeviceModelManifest,
    /**
     * The DevicePersonalizedInfo model constructor.
     * @property {module:model/DevicePersonalizedInfo}
     */
    DevicePersonalizedInfo: DevicePersonalizedInfo,
    /**
     * The DeviceStatePatchesStatePatch model constructor.
     * @property {module:model/DeviceStatePatchesStatePatch}
     */
    DeviceStatePatchesStatePatch: DeviceStatePatchesStatePatch,
    /**
     * The DevicesAddLabelRequest model constructor.
     * @property {module:model/DevicesAddLabelRequest}
     */
    DevicesAddLabelRequest: DevicesAddLabelRequest,
    /**
     * The DevicesAddNicknameRequest model constructor.
     * @property {module:model/DevicesAddNicknameRequest}
     */
    DevicesAddNicknameRequest: DevicesAddNicknameRequest,
    /**
     * The DevicesAddNicknameResponse model constructor.
     * @property {module:model/DevicesAddNicknameResponse}
     */
    DevicesAddNicknameResponse: DevicesAddNicknameResponse,
    /**
     * The DevicesListResponse model constructor.
     * @property {module:model/DevicesListResponse}
     */
    DevicesListResponse: DevicesListResponse,
    /**
     * The DevicesPatchStateRequest model constructor.
     * @property {module:model/DevicesPatchStateRequest}
     */
    DevicesPatchStateRequest: DevicesPatchStateRequest,
    /**
     * The DevicesPatchStateResponse model constructor.
     * @property {module:model/DevicesPatchStateResponse}
     */
    DevicesPatchStateResponse: DevicesPatchStateResponse,
    /**
     * The DevicesRemoveLabelRequest model constructor.
     * @property {module:model/DevicesRemoveLabelRequest}
     */
    DevicesRemoveLabelRequest: DevicesRemoveLabelRequest,
    /**
     * The DevicesRemoveNicknameRequest model constructor.
     * @property {module:model/DevicesRemoveNicknameRequest}
     */
    DevicesRemoveNicknameRequest: DevicesRemoveNicknameRequest,
    /**
     * The DevicesRemoveNicknameResponse model constructor.
     * @property {module:model/DevicesRemoveNicknameResponse}
     */
    DevicesRemoveNicknameResponse: DevicesRemoveNicknameResponse,
    /**
     * The Event model constructor.
     * @property {module:model/Event}
     */
    Event: Event,
    /**
     * The EventCommandPatch model constructor.
     * @property {module:model/EventCommandPatch}
     */
    EventCommandPatch: EventCommandPatch,
    /**
     * The EventsDeleteAllRequest model constructor.
     * @property {module:model/EventsDeleteAllRequest}
     */
    EventsDeleteAllRequest: EventsDeleteAllRequest,
    /**
     * The EventsListResponse model constructor.
     * @property {module:model/EventsListResponse}
     */
    EventsListResponse: EventsListResponse,
    /**
     * The EventsRecordDeviceEventsRequest model constructor.
     * @property {module:model/EventsRecordDeviceEventsRequest}
     */
    EventsRecordDeviceEventsRequest: EventsRecordDeviceEventsRequest,
    /**
     * The Invitation model constructor.
     * @property {module:model/Invitation}
     */
    Invitation: Invitation,
    /**
     * The JsonObject model constructor.
     * @property {module:model/JsonObject}
     */
    JsonObject: JsonObject,
    /**
     * The JsonValue model constructor.
     * @property {module:model/JsonValue}
     */
    JsonValue: JsonValue,
    /**
     * The LocalAccessEntry model constructor.
     * @property {module:model/LocalAccessEntry}
     */
    LocalAccessEntry: LocalAccessEntry,
    /**
     * The LocalAccessInfo model constructor.
     * @property {module:model/LocalAccessInfo}
     */
    LocalAccessInfo: LocalAccessInfo,
    /**
     * The ModelManifest model constructor.
     * @property {module:model/ModelManifest}
     */
    ModelManifest: ModelManifest,
    /**
     * The ModelManifestsListResponse model constructor.
     * @property {module:model/ModelManifestsListResponse}
     */
    ModelManifestsListResponse: ModelManifestsListResponse,
    /**
     * The ModelManifestsValidateCommandDefsRequest model constructor.
     * @property {module:model/ModelManifestsValidateCommandDefsRequest}
     */
    ModelManifestsValidateCommandDefsRequest: ModelManifestsValidateCommandDefsRequest,
    /**
     * The ModelManifestsValidateCommandDefsResponse model constructor.
     * @property {module:model/ModelManifestsValidateCommandDefsResponse}
     */
    ModelManifestsValidateCommandDefsResponse: ModelManifestsValidateCommandDefsResponse,
    /**
     * The ModelManifestsValidateComponentsRequest model constructor.
     * @property {module:model/ModelManifestsValidateComponentsRequest}
     */
    ModelManifestsValidateComponentsRequest: ModelManifestsValidateComponentsRequest,
    /**
     * The ModelManifestsValidateComponentsResponse model constructor.
     * @property {module:model/ModelManifestsValidateComponentsResponse}
     */
    ModelManifestsValidateComponentsResponse: ModelManifestsValidateComponentsResponse,
    /**
     * The ModelManifestsValidateDeviceStateRequest model constructor.
     * @property {module:model/ModelManifestsValidateDeviceStateRequest}
     */
    ModelManifestsValidateDeviceStateRequest: ModelManifestsValidateDeviceStateRequest,
    /**
     * The ModelManifestsValidateDeviceStateResponse model constructor.
     * @property {module:model/ModelManifestsValidateDeviceStateResponse}
     */
    ModelManifestsValidateDeviceStateResponse: ModelManifestsValidateDeviceStateResponse,
    /**
     * The PackageDef model constructor.
     * @property {module:model/PackageDef}
     */
    PackageDef: PackageDef,
    /**
     * The PlacesHints model constructor.
     * @property {module:model/PlacesHints}
     */
    PlacesHints: PlacesHints,
    /**
     * The Room model constructor.
     * @property {module:model/Room}
     */
    Room: Room,
    /**
     * The StateDef model constructor.
     * @property {module:model/StateDef}
     */
    StateDef: StateDef,
    /**
     * The AclEntriesApi service constructor.
     * @property {module:api/AclEntriesApi}
     */
    AclEntriesApi: AclEntriesApi,
    /**
     * The AdaptersApi service constructor.
     * @property {module:api/AdaptersApi}
     */
    AdaptersApi: AdaptersApi,
    /**
     * The CommandsApi service constructor.
     * @property {module:api/CommandsApi}
     */
    CommandsApi: CommandsApi,
    /**
     * The DevicesApi service constructor.
     * @property {module:api/DevicesApi}
     */
    DevicesApi: DevicesApi,
    /**
     * The EventsApi service constructor.
     * @property {module:api/EventsApi}
     */
    EventsApi: EventsApi,
    /**
     * The ModelManifestsApi service constructor.
     * @property {module:api/ModelManifestsApi}
     */
    ModelManifestsApi: ModelManifestsApi
  };

  return exports;
}));
